server/init.lua

```diff
@@ -52,26 +52,39 @@ local function updateMachineList(_, address, _)
 end
 event.listen("component_added", updateMachineList)
 
+local function identifyAsMainServer(_, _, sender, port, _, messageName)
+    if port == portNumber and messageName == "are_you_the_main_server" then
+        modem.send(sender, portNumber, "I_am_the_main_server")
+    end
+end
 if serverData.isMain == nil then
     -- Server not configured yet
-    local function filter(eventName, _, _, _, port, _, messageName)
-        return eventName == "modem_message" and port == portNumber and messageName == "I_am_the_main_server"
+
+    -- In case there's no response, server is main
+    serverData.isMain = true
+
+    local function detectMainServer(_, _, _, port, _, messageName)
+        if port == portNumber and messageName == "I_am_the_main_server" then
+            serverData.isMain = false
```

Comment:
Consider calling _save()_ after this.

Granted, it *is* called in the update loop, but if you don't run the process (or save manually), it will not save it to file and has to re-detect on reboot.

Reply:
It is saved after the timeout on line 85

Reply:
Ok